music_this_week_app/backend/spotifyHandler.py

```diff
@@ -86,13 +95,27 @@ def filter_artist(self, artist_name):
                 if a['name'] == artist_name:
                     if VERBOSE:
                         print("Exact match found!")
-                    return a['uri']
+                    artist_uri = a['uri']
+                    self.save_new_artist(artist_name, artist_uri)
+                    return artist_uri 
             # If there is no exact match, the first match is probably best.
-            return artists[0]['uri']
+            artist_uri = artists[0]['uri']           
+            self.save_new_artist(artist_name, artist_uri)
+            return artist_uri 
 
         # If we don't return in one of the If statements above, abort
         raise Exception('unexpected number of matches (%i) for artist %s' % (num_matches, artist))
 
+    def save_new_artist(self, artist_name, artist_uri):
+        """
+        saves artist name/uri pairs t DB
+        :param artist_name:
+        :param artist_uri:
+        :return None:
+        """
+        
+        artist = Artist.objects.get_or_create(spotify_uri = artist_uri, name = artist_name)
```

Comment:
Oh oops, I must have accidentally deleted artist.save() here when I was deleting print statements I'd used for testing